### DISCUSSION TITLE
Overriding Jupyter Default Theme

### DISCUSSION THREAD

**Discussion by srirag21** [jupyterlab/jupyterlab#16532] [General]
Hi all,

I am currently building and running a docker container that launches jupyter lab. I have set up a testpy package for my theme and locally I am able to launch jupyter lab and when I do so, it automatically applies the theme. However, the same is (at least sort of) not happening when I launch it through my container. I have scoured online and came across a few hopeful ideas like using the overrides.json file in order to override the default theme however these approaches have not worked for me. I was wondering if someone could point me to a simple method whether that's creating some sort of config file to specify the exact theme in order to ensure that upon launching the jupyter lab url it automatically applies it. Currently, I have it such that it shows the theme initially upon launch however a split second after, the jupyter lab page refreshes and goes back to the default theme. I can still click the settings tab and successfully activate my theme however my goal is to make it a one step process wherein I can simply launch the url with the theme attached when running the container. I would appreciate any help!

**Comment by krassowski** [comment]
Did you try `overrides.json`? https://jupyterlab.readthedocs.io/en/stable/user/directories.html#overrides-json

**Reply by srirag21** [reply]
Yep that's the current approach I'm using. One important thing I forgot to mention is that it seems like index.css is being taken into account and even when it goes back to the default theme, an image I had added on index.css still remains on the default theme. However, my changes to variables.css are not taken into account.

**Reply by srirag21** [reply]
My console looks like:
Unchecked runtime.lastError: The message port closed before a response was received.
jlab_core.c1feae0690cd2229a88c.js?v=c1feae0690cd2229a88c:1 

Language pack 'en_US' not installed!
fetch @ jlab_core.c1feae0690cd2229a88c.js?v=c1feae0690cd2229a88c:1

JupyterLab extension <my_theme> is activated!
821.a2a81361bc4cb2cbac34.js?v=a2a81361bc4cb2cbac34:1 

so it says it is activated and the image through the theme is there however when I go to Settings->Theme it says JupyterLab Light and not my custom theme. My custom theme is an option it's just that it's not selected

**Reply by srirag21** [reply]
Ah ok figured it out. For reference, if anyone is planning to run all this through a docker container: 
My environment was set up such that I was creating a virtual env and installing jupyter lab through it. If you are facing the same issues I was then I would highly suggest to go into the docker container using exec and typing jupyter lab path to figure out where you need to copy over the overrides.json file. You are most likely facing a path issue.

**Comment by srirag21** [comment]
As a follow up to this discussion. I did manage to get the theme to show however upon loading, an error box pops up with 
Error Loading Theme
Neither theme <theme name> nor default <theme name> loaded.

This means I did manage to make the default theme my theme and the theme does display and work as intended however the error box is extremely inconvenient and I would like to get rid of it. Has anyone faced this issue before, how would you suggest I go about fixing it?
